src/main/java/space/obminyashka/items_exchange/controller/UserController.java

```diff
@@ -219,4 +218,16 @@ private User getUser(String username) {
         return userService.findByUsernameOrEmail(username).orElseThrow(() -> new UsernameNotFoundException(
                 getMessageSource(ResponseMessagesHandler.ExceptionMessage.USER_NOT_FOUND)));
     }
+
+    private User findUserByValidEmail(Authentication authentication, String email) throws DataConflictException {
```

Comment:
That method ruins the Single responsibility principle - it does more than 1 general operation and should be split into a few others:
1. It validates email
2. It returns a user

Reply:
I splited this method